File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        st, end = 0, len(nums) - 1
        while st < end:
            mid = st + (end - st) // 2

            if nums[mid] > nums[end]:
                st = mid + 1
            else:
                end = mid

        return nums[st]

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int st=0,end=nums.size()-1;
        int ans=INT_MAX;
        while(st<=end){
            if (nums[st] <= nums[end]) {
                ans = min(ans, nums[st]);
                break;
            }
            int mid=st+(end-st)/2;
            if(nums[mid]>=nums[st] ){
                ans = min(ans, nums[st]);
                st=mid+1;
            }else{
                ans=min(ans,nums[mid]);
                end=mid-1;

            }
        }
        return ans;
    }
};